src/sentry/search/snuba/backend.py

```diff
@@ -508,6 +508,7 @@ def _get_queryset_conditions(
     ) -> Mapping[str, Condition]:
         queryset_conditions: Dict[str, Condition] = {
             "status": QCallbackCondition(lambda statuses: Q(status__in=statuses)),
+            "substatus": QCallbackCondition(lambda substatuses: Q(substatus__in=substatuses)),
```

Comment:
Fyi, I think this missing meant we weren't actually building up the SQL query properly to filter on `where substatus=<value>`. This should fix that.